consoles/serial_screen.pm

```diff
@@ -168,9 +161,7 @@ An undefined timeout will cause to wait indefinitely. A timeout of 0 means to
 just read once.
 
 =cut
-sub do_read
-{
-    my ($self, undef, %args) = @_;
+sub do_read ($self, undef, %args) {
```

Comment:
This one looks quite strange (include the previous version). It is called like this:

```
        my $read = $self->do_read($buf, max_size => $buflen / 2, timeout => remaining($sttime, $timeout));
        next READ unless (defined($read));
```

So `$buf` is basically ignored. Maybe we should get rid of the parameter in both functions and the invocation.

Reply:
For now I changed the second parameter to be `$` to ignore. I will look into your suggestion